core/src/main/java/org/apache/accumulo/core/spi/common/ContextClassLoaderFactory.java

```diff
@@ -57,4 +57,11 @@ public interface ContextClassLoaderFactory {
    */
   ClassLoader getClassLoader(String contextName);
 
+  /**
+   * Pass the service environment to allow for additional class loader configuration
+   *
+   * @param env the class loader environment
+   */
+  default void setEnvironment(ContextClassLoaderEnvironment env) {}
```

Comment:
A lot of the spi interfaces have a common pattern of having an init method that a takes a single InitParameters interface like [here](https://github.com/apache/accumulo/blob/2f27dbd188ea5cee32f486599b97181b136fa787/core/src/main/java/org/apache/accumulo/core/spi/compaction/CompactionPlanner.java#L72) 

```suggestion
  default void init(InitParameters params) {}
```

This could follow that pattern.  Also the init method name is good if its expected be called before anything else and once.

Reply:
I don't mind the name change for the method, but the `InitParameters` doesn't seem to be as accurate as the `...Environment` name. `InitParameters` implies to me that there are fixed params at initialization. I think that's a bit misleading. What's actually being passed is essentially `Supplier<ServiceEnvironment.Configuration>`, which isn't a fixed set of parameters, but a way to peek into the current system configuration, as needed.

If it's called `InitParameters`, and the method is still `getConfiguration()`, then that seems to imply a fixed configuration at initialization, when really the only thing that's fixed is the mechanism to get the configuration contained within the environment, not the configuration itself. So, if it's called `InitParameters`, I think the method would need to be changed to something like `.environmentConfigurationSupplier()`, so it's clear that the *supplier* is the fixed parameter, and the configuration is part of the environmental state, not just the state at initialization.

In putting together my suggested changes in ivakegg/accumulo#5, I considered suggesting getting rid of the ContextClassLoaderEnvironment entirely, and replace it with a `Supplier<ServiceEnvironment.Configuration>`, but that seemed too limiting, in case we want to expose more of the environment. Ideally, we'd pass the entire `ServiceEnvironment`, instead of just the configuration, but I explained why that was a problem in ivakegg/accumulo#5.

Reply:
That's a long way of saying, I think `init(ContextClassLoaderEnvironment env)` might be the best option, with the configuration retrievable via `env.getConfiguration()`.

Reply:
I tried to make a comment on this and it kept failing.  So I tried it at the top level and that worked.  I think GH was having problems.